tchannel-core/src/main/java/com/uber/tchannel/api/TChannel.java

```diff
@@ -85,6 +102,44 @@ private TChannel(Builder builder) {
         this.maxQueuedRequests = builder.maxQueuedRequests;
     }
 
+    private <T, U> ListenableFuture<Response<T>> callWithEncoding(
+            InetAddress host,
+            int port,
+            Request<U> request,
+            final Class<T> responseType,
+            ArgScheme scheme
+    ) throws InterruptedException {
+
+        RawRequest rawRequest = new RawRequest(
+                0,
```

Comment:
Let's get rid of hacking in the ID. Can we move this to the expectRequest method of the ResponseRouter?